components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
 	}
 
 	/**
-	 * Returns the blacklist per API
+	 * Returns the count per app per api
 	 *
 	 * {empty request body}
 	 */
 	@GET
-	@Path("/GetBlacklistPerApi/{apiId}")
-	@Consumes("application/json")
-	@Produces("text/plain")
-	public Response getBlacklistPerApi(@PathParam("apiId") String apiId) {
-		LOG.debug("getBlacklistPerApi Triggerd  apiName :" + apiId);
+	@Path("/count/{app}/{api}")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.TEXT_PLAIN)
+	public Response getBlacklistPerApi(@PathParam("app") String app, @PathParam("api") String api,
+									   @QueryParam("action") String action, @QueryParam("sp") String sp) {
 
 		Gson gson = new Gson();
 		StringBuilder errorMSG = new StringBuilder();
 		errorMSG.append("{" + "\"Failed\":").append("{").append("\"messageId\":\"").append("Blacklist result")
 				.append("\",").append("\"text\":\"");
 		errorMSG.append("Blacklist numbers could not be retrieved").append("\"" + "}}");
 
-		if (apiId != null) {
+		app = replaceAllWithWildcard(app);
+		api = replaceAllWithWildcard(api);
+		sp = replaceAllWithWildcard(sp);
+
+		BlacklistWhitelistDTO dto = new BlacklistWhitelistDTO();
+		dto.setAppId(app);
+		dto.setApiID(api);
+		dto.setAction(action);
+		dto.setServiceProvider(sp);
 
 			try {
-				MSISDNSearchDTO searchDTO = new MSISDNSearchDTO();
-				//searchDTO.setApiName(apiId);
-				searchDTO.setApiID(apiId);
-				String[] blacklist = blackListWhiteListService.loadBlacklisted(searchDTO);
-				StringBuilder successMSG = new StringBuilder();
 
-				successMSG.append("{").append("\"Success\":").append("{").append("\"messageId\":\"")
-						.append("Blacklist result").append("\",");
-				successMSG.append("\"text\":\"").append("Blacklist numbers were retrieved from the system".intern())
-						.append("\",").append("\"variables\":");
-				successMSG.append(gson.toJson(blacklist)).append("}}");
+				BlacklistWhitelistCountResponseDTO responseDTO = blackListWhiteListService.count(dto);
+				StringBuilder successMSG = new StringBuilder();
 
-				return Response.status(Response.Status.OK).entity(successMSG.toString()).build();
+				return Response.status(Response.Status.OK).entity(gson.toJson(responseDTO)).build();
 			} catch (BusinessException e) {
-				return Response.status(Response.Status.BAD_REQUEST).entity(e.getErrorType()).build();
+				return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getErrorType()).build();
 			}
-		} else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(errorMSG.toString()).build();
-		}
+
 	}
 
-	/**
-	 * Removes from the blacklist
-	 *
-	 * { "apiName":"USSD" }
-	 */
-	@POST
-	@Path("/RemoveFromBlacklist/{MSISDN}")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response removeFromBlacklist(@PathParam("MSISDN") String msisdn, String jsonBody) throws SQLException {
-		LOG.debug("removeFromBlacklist Triggerd  jsonBody :" + jsonBody + " , msisdn :" + msisdn);
+	@GET
+	@Path("/downloadAsZip/{app}/{api}")
```

Comment:
avoid using camel cases in endpoint URLs (downloadAsZip). It should be hyphen character separated